mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt

```diff
@@ -25,33 +26,52 @@ import androidx.preference.EditTextPreferenceDialogFragmentCompat
 class CustomInputTypePreference constructor(context: Context, attrs: AttributeSet) :
     EditTextPreference(context, attrs) {
     private val inputType: Int
+    private var autofillHints: String? = null
 
     init {
         context.obtainStyledAttributes(attrs, intArrayOf(android.R.attr.inputType)).apply {
             inputType = getInt(0, 0)
             recycle()
         }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.obtainStyledAttributes(attrs, intArrayOf(android.R.attr.autofillHints)).apply {
+                autofillHints = getString(0)
+                recycle()
+            }
+        }
```

Comment:
Can't we unify those two blocks into one? The `attr` definition should be present no matter what the actual API level is.
I'd also do the parsing (string -> string array conversion) here and pass around the array, as the string format is specific to the attribute.

Also, `android:autoFillHints` is comma separated, not pipe separated: https://developer.android.com/reference/android/view/View.html#attr_android:autofillHints

Reply:
Done, but can you give it a try on api 22 emulators? The app freezes there if I open the settings activity.

Reply:
Doesn't seem to happen for me (?) I created a new Emulator for Nexus 7, using API 22 + Google APIs as emulator image. Then I started a fossBeta build in that emulator, and settings activity seems just fine for me.

Reply:
It seems that there was an issue with the emulator itself, not the build.